insights/client/crypto.py

```diff
@@ -0,0 +1,122 @@
+import os.path
+import logging
+import shutil
+import tempfile
+import shlex
+import subprocess
+
+from .constants import InsightsConstants as constants
+
+logger = logging.getLogger(__file__)
+
+
+def _call_gpg(home, command):
+    """
+    Run /usr/bin/gpg.
+
+    Parameters:
+        home (str): Path to GPG's home directory.
+        command (str): Actual arguments passed to the GPG binary.
+
+    Returns (str, str, int):
+        Standard output, standard error and the return code.
+    """
+    command = (
+        "/usr/bin/gpg --homedir {home} {command}"
+    ).format(home=home, command=command)
+    process = subprocess.Popen(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    raw_stdout, raw_stderr = process.communicate()
+    stdout, stderr = raw_stdout.decode("utf-8"), raw_stderr.decode("utf-8")
+
+    logger.debug(command)
+    logger.debug("status code {code}".format(code=process.returncode))
+
+    if process.returncode > 0 and stdout:
+        logger.debug("stdout is not empty:\n%s", stdout.strip())
+    if process.returncode > 0 and stderr:
+        logger.debug("stderr is not empty:\n%s", stderr.strip())
+
+    return stdout, stderr, process.returncode
+
+
+def verify_gpg_signed_file(file, signature):
```

Comment:
I would still pass in the key to use here.  It makes it easier to test and makes the function more flexible in case the code needs to use different keys in the future.